scripts/travis-emscripten/build_emscripten.sh

```diff
@@ -42,6 +42,10 @@ fi
 
 WORKSPACE=/root/project
 
+# Increase nodejs stack size
+sed -i -e 's/NODE_JS="nodejs"/NODE_JS=["nodejs", "--stack_size=8192"]/' /root/.emscripten
```

Comment:
I'm not familiar with the emscripten compilation process but what is `node` used for there and what was failing without the stack size increase?

Reply:
For the record: https://travis-ci.org/ethereum/solidity/jobs/338054884

```
Linking CXX executable soljson.js
/emsdk_portable/emscripten/tag-1.35.4/tools/js-optimizer.js:7758
    print(JSON.stringify(ast));
               ^
RangeError: Maximum call stack size exceeded
fs.js:436
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^
Error: ENOENT, no such file or directory '/tmp/tmpEOtDSr.jsfunc_7.js'
    at Object.fs.openSync (fs.js:436:18)
    at Object.fs.readFileSync (fs.js:290:15)
    at read (/emsdk_portable/emscripten/tag-1.35.4/tools/js-optimizer.js:68:34)
    at Object.<anonymous> (/emsdk_portable/emscripten/tag-1.35.4/tools/js-optimizer.js:7708:11)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
```

So it seems the output code AST became too big, but at least the resulting .js file is still only 7.5MB.

Reply:
It looks like emscripten has an optimizer written in javascript which is run via node.